[scroll-animations] Combine named timeline maps on AnimationTimelinesController

nmoucht opened the pull request:
#### 7ed88be64a1c8c8ca12c99af4bec06c8b498132d
<pre>
[scroll-animations] Combine named timeline maps on AnimationTimelinesController
<a href="https://bugs.webkit.org/show_bug.cgi?id=281708">https://bugs.webkit.org/show_bug.cgi?id=281708</a>
<a href="https://rdar.apple.com/138153691">rdar://138153691</a>

Reviewed by Antoine Quint.

Combine the scroll maps for ScrollTimeline and ViewTimeline to one map.

* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-timeline-named-scroll-progress-timeline.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/scroll-timeline-range-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-dynamic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-inset-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-lookup-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-name-shadow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-range-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-used-values-expected.txt:
* Source/WebCore/animation/AnimationTimelinesController.cpp:
(WebCore::originatingElement):
(WebCore::AnimationTimelinesController::timelinesForName):
(WebCore::AnimationTimelinesController::registerNamedScrollTimeline):
(WebCore::AnimationTimelinesController::registerNamedViewTimeline):
(WebCore::AnimationTimelinesController::unregisterNamedTimeline):
(WebCore::AnimationTimelinesController::timelineForName const):
(WebCore::AnimationTimelinesController::unregisterNamedScrollTimeline): Deleted.
(WebCore::AnimationTimelinesController::scrollTimelineForName const): Deleted.
(WebCore::AnimationTimelinesController::unregisterNamedViewTimelineForSubject): Deleted.
(WebCore::AnimationTimelinesController::viewTimelineForNameAndSubject const): Deleted.
* Source/WebCore/animation/AnimationTimelinesController.h:
* Source/WebCore/animation/CSSAnimation.cpp:
(WebCore::CSSAnimation::syncPropertiesWithBackingAnimation):
* Source/WebCore/animation/ScrollTimeline.cpp:
(WebCore::ScrollTimeline::setSource):
* Source/WebCore/animation/ScrollTimeline.h:
* Source/WebCore/animation/ViewTimeline.cpp:
(WebCore::ViewTimeline::setSubject):
* Source/WebCore/animation/ViewTimeline.h:
* Source/WebCore/style/Styleable.cpp:
(WebCore::Styleable::updateCSSScrollTimelines const):
(WebCore::Styleable::updateCSSViewTimelines const):

Canonical link: <a href="https://commits.webkit.org/285668@main">https://commits.webkit.org/285668@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cc7529225e142b65a360f0397f6865743c073e44

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73460 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52889 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26268 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77724 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24698 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/75575 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62020 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/674 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57727 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16168 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76527 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/47797 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63200 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38138 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44410 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20684 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23029 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/66215 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21036 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79343 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/776 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/270 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66124 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/917 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63213 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9252 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11314 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/739 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/3476 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/769 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/754 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/773 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,53 +311,77 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
-        auto newScrollTimeline = ScrollTimeline::create(name, axis);
-        newScrollTimeline->setSource(&source);
-        m_nameToScrollTimelineMap.set(name, WTFMove(newScrollTimeline));
+    auto* element = &matchElement;
```
`RefPtr`

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,53 +311,77 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
 {
-    m_nameToScrollTimelineMap.remove(name);
+    // https://drafts.csswg.org/scroll-animations-1/#timeline-scoping
+    // From the spec: A named scroll progress timeline or view progress timeline is referenceable by:
```
I think you should remove "From the spec: ".

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,53 +311,77 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
+
+    m_nameToTimelineMap.ensure(name, [] {
+        return Vector<NamedTimelineMapEntry> { };
+    }).iterator->value.append({ newScrollTimeline, Ref<Element> { source } });
```
Is this correct? This will replace the timeline instance on all calls, which may break the relationship established with previously instantiated CSS Animations. The previous method would update the entry found in the map.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -68,21 +68,26 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
+    NamedTimelineMapEntry* determineTreeOrder(Vector<NamedTimelineMapEntry>, Element&) const;
+
+    UncheckedKeyHashMap<AtomString, Vector<NamedTimelineMapEntry>> m_nameToTimelineMap;
+
```
No need for this final empty line.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -68,21 +68,26 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
+    struct NamedTimelineMapEntry {
+        Ref<AnimationTimeline> timeline;
+        Ref<Element> element;
+    };
```
I'm not sure this is entirely necessary, we could simply store a `Vector<ViewTimeline>` and then look the subject up via the subject associated with the timeline.

graouts replied:
Also seems like a bad idea to use `Ref<Element>`, we would probably want a WeakPtr there so as not to keep the subject alive.

nmoucht replied:
Made this just hold a Ref<AnimationTimeline>.

graouts commented on LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-timeline-none-expected.txt:
```diff
@@ -1,4 +1,4 @@
 
 FAIL Animation with animation-timeline:none holds current time at zero assert_equals: expected "100px" but got "0px"
-FAIL Animation with unknown timeline name holds current time at zero assert_equals: expected "100px" but got "101.9375px"
+FAIL Animation with unknown timeline name holds current time at zero assert_equals: expected "100px" but got "100.203125px"
```
This is just a flaky result, no need to commit this.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -68,21 +68,21 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
-    void unregisterNamedViewTimelineForSubject(const AtomString&, const Element&);
-    ViewTimeline* viewTimelineForNameAndSubject(const AtomString&, const Element&) const;
+    void unregisterNamedTimeline(const AtomString&, Element&);
+    RefPtr<AnimationTimeline> timelineForName(const AtomString&, Element&) const;
```
The `Element&` parameter should be `const`.

graouts replied:
I'm not sure returning a `RefPtr` is correct either. I believe it's up to the call site to protect that value.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -68,21 +68,21 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
     void maybeClearCachedCurrentTime();
 
+    RefPtr<AnimationTimeline> timeline(Vector<Ref<AnimationTimeline>>, Element&) const;
+    RefPtr<AnimationTimeline> determineTreeOrder(Vector<Ref<AnimationTimeline>>, Element&) const;
```
These should use `const` for the `Element` parameter I believe, and I think they should return a raw pointer.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -68,21 +68,21 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
     void registerNamedViewTimeline(const AtomString&, Element&, ScrollAxis, ViewTimelineInsets&&);
-    void unregisterNamedViewTimelineForSubject(const AtomString&, const Element&);
-    ViewTimeline* viewTimelineForNameAndSubject(const AtomString&, const Element&) const;
+    void unregisterNamedTimeline(const AtomString&, Element&);
```
`const Element&`.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -68,21 +68,21 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
     void cacheCurrentTime(ReducedResolutionSeconds);
     void maybeClearCachedCurrentTime();
 
+    RefPtr<AnimationTimeline> timeline(Vector<Ref<AnimationTimeline>>, Element&) const;
```
I think I'd just do away with that method and just have it inside `timelineForName` as it appears to have a single call site. If not, it should be `static` and ideally have a less generic name.

nmoucht replied:
No longer relevant for this patch.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -355,35 +393,21 @@ void AnimationTimelinesController::registerNamedViewTimeline(const AtomString& n
-        m_nameToViewTimelinesMap.remove(it);
+    auto it = m_nameToTimelineMap.find(name);
+    if (it != m_nameToTimelineMap.end()) {
+        it->value.removeFirstMatching([&] (const Ref<AnimationTimeline> entry) {
```
You can use `auto& timeline` here.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,42 +311,79 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
 
     if (existingTimelineIndex != notFound) {
-        auto& existingViewTimeline = timelines[existingTimelineIndex];
+        auto existingViewTimeline = downcast<ViewTimeline>(timelines[existingTimelineIndex]);
```
I believe this creates a second `Ref`, please use a reference like the code did before.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimeline.h:
```diff
@@ -63,6 +64,7 @@ class AnimationTimeline : public RefCounted<AnimationTimeline>, public CanMakeWe
     virtual void suspendAnimations();
     virtual void resumeAnimations();
     bool animationsAreSuspended() const;
+    virtual Element* originatingElement() const { return nullptr; };
```
I'm not a fan of this new method. I understand the idea here but we still use it a lot in conjunction with a type check. At the moment, only scroll and view timelines can be instantiated via CSS in the wrong place, so it seems a bit generic to put it on `AnimationTimeline`. I'd just do away with it entirely and just downcast in the places where we must check this and use `source()` and `subject()` accordingly… or introduce that method on `ScrollTimeline`.

nmoucht replied:
Added static function for this.

graouts reviewed (changes requested):
Generally, I think it would make more sense to split this patch in two tasks. First, generalize the name+element pairing to scroll timelines as well and use a single map. Second, introduce the tree order code, possibly along with a `timeline-scope` patch.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,75 +311,76 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
+    });
+
+    if (existingTimelineIndex != notFound) {
+        auto& existingScrollTimeline = downcast<ScrollTimeline>(timelines[existingTimelineIndex].get());
```
`Ref`

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,75 +311,76 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
-        auto& existingViewTimeline = timelines[existingTimelineIndex];
-        existingViewTimeline->setAxis(axis);
-        existingViewTimeline->setInsets(WTFMove(insets));
+        auto& existingViewTimeline = downcast<ViewTimeline>(timelines[existingTimelineIndex].get());
```
`Ref`

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,75 +311,76 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
-    if (timelines.isEmpty())
-        m_nameToViewTimelinesMap.remove(it);
+    auto it = m_nameToTimelineMap.find(name);
+    if (it != m_nameToTimelineMap.end()) {
```
I prefer the early return approach as used before.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/style/Styleable.cpp:
```diff
@@ -893,7 +893,7 @@ void Styleable::updateCSSScrollTimelines(const RenderStyle* currentStyle, const
         for (auto& previousTimelineName : currentStyle->scrollTimelineNames()) {
             if (!currentTimelineNames.contains(previousTimelineName))
-                timelinesController->unregisterNamedScrollTimeline(previousTimelineName);
+                timelinesController->unregisterNamedTimeline(previousTimelineName, element);
```
I wonder whether this is correct. Do we need to keep a separate method for `ScrollTimeline` and `ViewTimeline` so that we look up the timeline to remove by type? Is there any provision made in the spec for `scroll-timeline-name` and `view-timeline-name` to have exclusive names?

nmoucht replied:
From the spec:
> In case of a name conflict on the same element, names declared later in the naming property ([scroll-timeline-name](https://drafts.csswg.org/scroll-animations-1/#propdef-scroll-timeline-name), [view-timeline-name](https://drafts.csswg.org/scroll-animations-1/#propdef-view-timeline-name)) take precedence, and [scroll progress timelines](https://drafts.csswg.org/scroll-animations-1/#scroll-progress-timelines) take precedence over [view progress timelines](https://drafts.csswg.org/scroll-animations-1/#view-progress-timelines).

I think based on this we should never have an element with more than one timeline associated with the same name (although we will have to implement this logic in a later patch).

graouts reviewed (changes requested):
The source changes look OK aside from the unregistration call, a minor stylistic nit and a couple like Safer C++ warnings… but I'm worried about the many new `FAIL` results. Those need to be looked at carefully.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
 }
 
-void AnimationTimelinesController::registerNamedScrollTimeline(const AtomString& name, Element& source, ScrollAxis axis)
+static Element* originatingElement(Ref<AnimationTimeline> timeline)
```
This should take in a `Ref<ScrollTimeline>&`.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.h:
```diff
@@ -67,22 +67,20 @@ class AnimationTimelinesController final : public CanMakeCheckedPtr<AnimationTim
 
+    Vector<Ref<AnimationTimeline>>& timelinesForName(const AtomString&);
+
+    UncheckedKeyHashMap<AtomString, Vector<Ref<AnimationTimeline>>> m_nameToTimelineMap;
```
This should be a `Vector<Ref<ScrollTimeline>>` I believe, since we're not going to register any timeline that isn't a scroll timeline.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
 {
-    m_nameToScrollTimelineMap.remove(name);
+    return m_nameToTimelineMap.ensure(name, [] {
+        return Vector<Ref<AnimationTimeline>> { };
```
`Ref<ScrollTimeline>`.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
+
+    auto existingTimelineIndex = timelines.findIf([&](auto& timeline) {
+        return is<ScrollTimeline>(timeline) && originatingElement(timeline) == &source;
+    });
```
Shouldn't this check this is a `ScrollTimeline` and _not_ a `ViewTimeline`? Once we only store `ScrollTimeline` you can only check that it's not a `ViewTimeline`.

nmoucht replied:
Fixed.

graouts commented on Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
```diff
@@ -346,8 +346,8 @@ struct PlatformIntelligenceTextEffectID: Hashable {
-        wrappedDestinationEffect.isDestination = true
-        wrappedDestinationEffect.animateRemovalWhenDone = true
+//        wrappedDestinationEffect.isDestination = true
+//        wrappedDestinationEffect.animateRemovalWhenDone = true
```
This doesn't belong in this patch.

nmoucht replied:
Removed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
 }
 
-void AnimationTimelinesController::registerNamedScrollTimeline(const AtomString& name, Element& source, ScrollAxis axis)
+static Element* originatingElement(Ref<ScrollTimeline> timeline)
```
This could be a `Ref<ScrollTimeline>&`. But really I think we can do away with this static method entirely, it only really needs to be used once in `unregisterNamedTimeline()` since the other call sites are places where we're already dealing with logic specific to `ScrollTimeline` or `ViewTimeline`.

nmoucht replied:
Removed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
-        newScrollTimeline->setSource(&source);
-        m_nameToScrollTimelineMap.set(name, WTFMove(newScrollTimeline));
-    }
+    return is<ViewTimeline>(timeline) ? downcast<ViewTimeline>(timeline)->subject() : timeline->source();
```
It's more efficient to use `dynamicDowncast<>` here.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -355,35 +360,30 @@ void AnimationTimelinesController::registerNamedViewTimeline(const AtomString& n
-    if (timelines.isEmpty())
-        m_nameToViewTimelinesMap.remove(it);
+    if (it->value.isEmpty())
+        m_nameToTimelineMap.remove(name);
```
I expect `m_nameToTimelineMap.remove(it)` is more efficient.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -355,35 +360,30 @@ void AnimationTimelinesController::registerNamedViewTimeline(const AtomString& n
+    if (it == m_nameToTimelineMap.end())
         return;
-
-    auto& timelines = it->value;
```
I would have kept this line and avoid using `it->value` twice. Makes it clearer what we're manipulating as well. It matches `timelineForName()` as well.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/AnimationTimelinesController.cpp:
```diff
@@ -30,6 +30,7 @@
 #include "CSSTransition.h"
 #include "Document.h"
 #include "DocumentTimeline.h"
+#include "ElementInlines.h"
```
What is this for?

nmoucht replied:
Removed.

nmoucht committed:
[scroll-animations] Combine named timeline maps on AnimationTimelinesController
https://bugs.webkit.org/show_bug.cgi?id=281708
rdar://138153691

Reviewed by Antoine Quint.

Combine the scroll maps for ScrollTimeline and ViewTimeline to one map.

* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-timeline-named-scroll-progress-timeline.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/scroll-timeline-range-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/timeline-scope-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-dynamic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-inset-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-lookup-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-name-shadow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-range-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/view-timeline-used-values-expected.txt:
* Source/WebCore/animation/AnimationTimelinesController.cpp:
(WebCore::originatingElement):
(WebCore::AnimationTimelinesController::timelinesForName):
(WebCore::AnimationTimelinesController::registerNamedScrollTimeline):
(WebCore::AnimationTimelinesController::registerNamedViewTimeline):
(WebCore::AnimationTimelinesController::unregisterNamedTimeline):
(WebCore::AnimationTimelinesController::timelineForName const):
(WebCore::AnimationTimelinesController::unregisterNamedScrollTimeline): Deleted.
(WebCore::AnimationTimelinesController::scrollTimelineForName const): Deleted.
(WebCore::AnimationTimelinesController::unregisterNamedViewTimelineForSubject): Deleted.
(WebCore::AnimationTimelinesController::viewTimelineForNameAndSubject const): Deleted.
* Source/WebCore/animation/AnimationTimelinesController.h:
* Source/WebCore/animation/CSSAnimation.cpp:
(WebCore::CSSAnimation::syncPropertiesWithBackingAnimation):
* Source/WebCore/animation/ScrollTimeline.cpp:
(WebCore::ScrollTimeline::setSource):
* Source/WebCore/animation/ScrollTimeline.h:
* Source/WebCore/animation/ViewTimeline.cpp:
(WebCore::ViewTimeline::setSubject):
* Source/WebCore/animation/ViewTimeline.h:
* Source/WebCore/style/Styleable.cpp:
(WebCore::Styleable::updateCSSScrollTimelines const):
(WebCore::Styleable::updateCSSViewTimelines const):

Canonical link: https://commits.webkit.org/285668@main
